client/lib/purchases/assembler.js

```diff
@@ -57,17 +56,17 @@ function createPurchaseObject( purchase ) {
 	};
 
 	if ( 'credit_card' === purchase.payment_type ) {
-		return merge( {}, object, {
-			payment: {
-				creditCard: {
-					id: Number( purchase.payment_card_id ),
-					type: purchase.payment_card_type,
-					number: Number( purchase.payment_details ),
-					expiryDate: purchase.payment_expiry,
-					expiryMoment: purchase.payment_expiry ? i18n.moment( purchase.payment_expiry, 'MM/YY' ) : null
-				}
+		const payment = Object.assign( {}, object.payment, {
+			creditCard: {
+				id: Number( purchase.payment_card_id ),
+				type: purchase.payment_card_type,
+				number: Number( purchase.payment_details ),
+				expiryDate: purchase.payment_expiry,
+				expiryMoment: purchase.payment_expiry ? i18n.moment( purchase.payment_expiry, 'MM/YY' ) : null
 			}
 		} );
+
+		return Object.assign( {}, object, { payment } );
```

Comment:
>Won't `Object.assign` use pull in the `O.assign` shim when you're already using `lodash/assign` in other places?

This is true, though speaks to a larger problem of inconsistency between using `Object.assign` and `_.assign` in the codebase. It's likely the shim has already been included in whichever bundle is making use of the assembler.